### PR TITLE
DRV-378: Support for 3rd party auth APIv4

### PR DESCRIPTION
**Pull request by parkhomenko**
[Notes]
Introduces auth functions for APIv4

[Jira ticket]
https://faunadb.atlassian.net/browse/DRV-378

### CODE REVIEW

File: FaunaDB.Client/Query/Language.Authentication.cs [hunk 1]
```diff
@@ -45,19 +45,91 @@ public partial struct Language
...
+        public static Expr AccessProvider(Expr name) =>
+            UnescapedObject.With("access_provider", name);
+        
+        /// <summary>
+        /// Creates a new AccessProvider expression.
+        /// <para>
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/accessprovider">docs</see>.
+        /// </para>
+        /// </summary>
+        public static Expr AccessProvider(Expr name, Expr scope) =>
```
Comment (marrony) [inline FaunaDB.Client/Query/Language.Authentication.cs:123]:
```suggestion
        public static Expr AccessProvider(Expr name, Expr scope = null) =>
```
and you can remove the `public static Expr AccessProvider(Expr name)` version

File: FaunaDB.Client.Test/ClientTest.cs [hunk 2]
```diff
@@ -2171,6 +2171,189 @@ public async Task TestCustomHttp()
...
+                    "jwks_uri", jwksUri,
+                    "roles", Arr(role)
+                )));
+
+            IReadOnlyList<RefV> roles = accessProvider.At("roles").Collect(Field.To<RefV>());
+            Assert.AreEqual(1, roles.Count);
+
+            Assert.AreEqual(
+                accessProviderName,
+                accessProvider.Get(Field.At("name")).To<string>().Value);
```
Comment (marrony) [inline FaunaDB.Client.Test/ClientTest.cs:2205]: extract those Fields into a variables, the top the class contains many other fields

File: FaunaDB.Client/Query/Language.Authentication.cs (hunk 1 above)
Comment (parkhomenko) [reply FaunaDB.Client/Query/Language.Authentication.cs:123]: removed this method, and added `null` as you suggested

File: FaunaDB.Client.Test/ClientTest.cs (hunk 2 above)
Comment (parkhomenko) [reply FaunaDB.Client.Test/ClientTest.cs:2205]: extracted all the fields I used in the tests, thanks

**Review by evbruno — APPROVED** [review]
lgtm

**Review by marrony — DISMISSED** [review]

**Comment by parkhomenko** [comment]
@marrony @evbruno 
could you re-approve this PR, I've merged changed from master into it and returned back fauna private image to make tests pass, since we don't have api v4 support for public fauna docker image

**Review by fireridlle — APPROVED** [review]

**Review by tigger9flow — APPROVED** [review]

**Pull request merged**